velox/connectors/hive/SplitReader.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/dwio/common/Reader.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::dwio::common {
+class BufferedInput;
+}
+
+namespace facebook::velox::connector::hive {
+
+class HiveConnectorSplit;
+class HiveTableHandle;
+class HiveColumnHandle;
+
+static const char* kPath = "$path";
+static const char* kBucket = "$bucket";
+
+class SplitReader {
+ public:
+  static std::unique_ptr<SplitReader> create(
+      std::shared_ptr<velox::connector::hive::HiveConnectorSplit> hiveSplit,
+      RowTypePtr readerOutputType,
+      std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
+          partitionKeys,
+      std::shared_ptr<common::ScanSpec> scanSpec,
+      memory::MemoryPool*& pool);
+
+  SplitReader(
+      std::shared_ptr<velox::connector::hive::HiveConnectorSplit> hiveSplit,
+      RowTypePtr readerOutputType,
+      std::unordered_map<std::string, std::shared_ptr<HiveColumnHandle>>&
+          partitionKeys,
+      std::shared_ptr<common::ScanSpec> scanSpec,
+      memory::MemoryPool*& pool);
+
+  virtual ~SplitReader() = default;
+
+  /**
```

Comment:
We use `///` for comments.

Reply:
I see other files mostly use "/** */ " for API comments, like in Reader.h, SeekableInputStream.h, IntDecoder.h, SelectiveColumnReader.h and many more, while some uses "//", e.g. ScanSpec.h, SelectiveColumnReader.h, not many files use "///". Is this a new thing? If we are switching from "/** */ " to "///", it may need some clarification and better to be in separate PRs.

Reply:
@yingsu00 you are right that some of the older Reader files still use the "/** / " style. But the coding style document explicitly says not to use them. We have been moving away from the old style to the new style whenever we touch a file with them.
``
Don't use old-style /* */ comments inside code or on top-level header comments. It adds two additional lines and makes headers more verbose than they need to be
``
https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md

Reply:
Yes let's use `///` for all public function comments